File: python/p_036.py
MaxNumber = 1000000

def is_symmetric(Number):
    Number_string = str(Number)
    return list(str(Number)) == list(reversed(Number_string))

Sum = 0
for N in range(MaxNumber + 1):
    if is_symmetric(N):
        if bin(N)[2:] == (bin(N)[2:])[::-1]:
            Sum += N

print(Sum)

File: julia/p_036.jl
MaxNumber = 1000000

function is_symmetric(Number::Int)
    Number_string = string(Number)
    return Number_string == reverse(Number_string)
end

Sum = 0
for N in 1:MaxNumber
    if is_symmetric(N)
        if digits(N, base=2) == digits(N, base=2) |> reverse
            Sum += N
        end
    end
end

println(Sum)